airflow/providers/snowflake/hooks/snowflake.py

```diff
@@ -142,3 +141,18 @@ def set_autocommit(self, conn, autocommit: Any) -> None:
 
     def get_autocommit(self, conn):
         return getattr(conn, 'autocommit_mode', False)
+
+    def run(self, sql, autocommit=False, parameters=None):
+        """
+        Snowflake-connector doesn't allow natively the execution of multiple SQL statements in the same
+        call. So for allowing to pass files or strings with several queries this method is coded,
+        that relies on run from DBApiHook
+        """
+        if isinstance(sql, str):
+            with closing(self.get_conn()) as conn:
+                if self.supports_autocommit:
```

Comment:
this generic `if self.supports_autocommit` logic makes sense in DbApiHook because it's a generic base class that means to support many different databases -- some which support autocommit and some which don't.  in your case here, you are implementing for one specific database and you _know_ whether it supports autocommit or not -- so you don't need to check --- just do the right thing.